napari/utils/events/event.py

```diff
@@ -763,7 +763,12 @@ def __call__(self, *args, **kwargs) -> Event:
                 self.disconnect(cb)
         finally:
             self._emitting = False
-            if event._pop_source() != self.source:
+            ps = event._pop_source()
+            # test for identity before equality is faster, and can
+            # shave a non-negligible amount of time.
+            if ps != self.source:
+                assert isinstance(ps, EventEmitter), ps
```

Comment:
Sorry, I wasn't clear, I still think that you were right and this should only test for identity, as we really want the source to be *exactly* the same object. But I would *also* leave the change to `__eq__`.

Reply:
```suggestion
            assert isinstance(ps, EventEmitter), ps
            if ps is not self.source:
```